src/reporter.js

```diff
@@ -33,28 +39,52 @@ class Reporter {
         this.timerSnapshot = null;      // timer that goes off when it's time to do a snapshot write
 
         this.stopped = false;           // true if this Reporter has been stopped
+
+        this.reportPath = "";           // path for the report folder default is same folder smashtest file
+        this.history = false;           // activate history for reporting
     }
 
     /**
-     * @return {String} The absolute path of the report html file
-     */
+    * @return {String} The absolute path of the report html file
+    */
     getFullReportPath() {
-        return path.join(process.cwd(), REPORT_FILENAME);
+        return path.join(folder, "report.html");
+    }
+
+    /**
+    * @return {String} The path for report folder
+    */
+   getPathFolder() {
+    initialFolder = (this.history === false ? "smashtest" : `${path.join(`smashtest`, `report`, `smashtest-${dateFormat}`)}`);
+    folder = (this.reportPath === "" ? initialFolder : `${path.join(this.reportPath, initialFolder)}`);
+    return folder;
+}
+
+    /**
+    * @return {String} Variable custom path for report folder
+    */
+    getCustomPath() {
+        return folder;
     }
 
     /**
      * Starts the reporter, which generates and writes to disk a new report once every REPORT_GENERATE_FREQUENCY ms
      */
     async start() {
         // Clear out existing screenshots (one by one)
+        reportFilename = path.join(folder, "report.html");
+        reportDataFilename = path.join(folder, "report-data.js");
+        passedDataFilename = path.join(folder, "passed-data");
```

Comment:
One last thing. I think `passed-data` should go directly under `smashtest/`, even if history is on. That's because it's not part of the report (it's used for the `-s` flag), and a current run has to access to the `passed-data` file from the previous run. In other words, there should only be one `passed-data` file.

Reply:
Actually, now that I think about it, maybe it's best to have two copies of `passed-data`, one in the dated report folder (like you have), and one directly under `smashtest/`. That way, you can run `smashtest` and it will look to `smashtest/passed-data`, but if you want to use a previous version you can use the one in the reports folder.

Reply:
Ok i add a passed-data directly on /smashtest if history is activated